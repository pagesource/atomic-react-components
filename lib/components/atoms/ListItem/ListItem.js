// @flow
/**
 *
 * ListItem
 *
 */
import React from 'react';
import styled from 'styled-components';

import styles from './ListItem.style';
import type { Props } from './types';

const ListItem = ({ className, children }: Props): Node => {
  const classes = ['item', className].join(' ');

  return <li className={classes}>{children}</li>;
};

ListItem.defaultProps = {
  className: '',
};

export default styled(ListItem)`
  ${styles};
`;

export { ListItem as ListItemVanilla };
