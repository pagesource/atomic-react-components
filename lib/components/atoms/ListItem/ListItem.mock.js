const defaultConfig = {};

export default defaultConfig;
