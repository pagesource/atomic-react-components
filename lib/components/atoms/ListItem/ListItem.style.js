import { css } from 'styled-components';

export default css`
  margin: 10px;
  ${props => (props.inheritedStyles ? props.inheritedStyles : '')};
`;
