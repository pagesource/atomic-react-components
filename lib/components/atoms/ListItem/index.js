// @flow
export { default } from './ListItem';
export { ListItemVanilla } from './ListItem';
