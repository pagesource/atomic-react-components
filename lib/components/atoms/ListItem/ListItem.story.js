import React from 'react';
import { storiesOf } from '@storybook/react';
// import { action } from '@storybook/addon-actions';
import defaultConfig from './ListItem.mock';

// Import Styled Component to showcase variations
import ListItem, { ListItemVanilla } from '.';

// TODO: indentation is not working fine in editor. Need to check for .hbs files
// $FlowFixMe eslint-disable-next-line extra-rules/no-commented-out-code
storiesOf('Atoms', module)
  .addParameters({ jest: ['ListItem', 'ListItemVanilla'] })
  .add('CTA Knobs', () => (
    <ListItemVanilla
      {...defaultConfig}
      className={`hide-default-sample ${defaultConfig.className}`}
    />
  ))
  .add('CTA', () => <ListItem {...defaultConfig} />);
