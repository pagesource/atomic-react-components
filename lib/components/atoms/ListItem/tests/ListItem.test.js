import { shallow } from 'enzyme';

import ListItem from '../index';

describe('<ListItem />', () => {
  let ListItemComponent = '';
  beforeEach(() => {
    ListItemComponent = shallow(<ListItem>Test</ListItem>);
  });

  test('should render correctly', () => {
    expect(ListItemComponent).toMatchSnapshot();
  });
});
