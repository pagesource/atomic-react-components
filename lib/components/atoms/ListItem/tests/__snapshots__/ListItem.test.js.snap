// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> should render correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "ListItem-qqbojw-0",
        "isStatic": false,
        "rules": Array [
          "margin:10px;",
          [Function],
          ";",
          ";",
        ],
      },
      "displayName": "ListItem",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "ListItem-qqbojw-0",
      "target": [Function],
      "toString": [Function],
      "usesTheme": false,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  Test
</StyledComponent>
`;
